.github/workflows/test.yaml

```
@@ -35,6 +35,12 @@ jobs:
       - name: Run make test
         run: |
           make test-unit
+      - name: Upload unit-test coverage reports to CodeCov # more at https://github.com/codecov/codecov-action
+        uses: codecov/codecov-action@v3
+        with:
+          flags: unit
+          fail_ci_if_error: true
+          verbose: true
```

```suggestion
          verbose: true
          token: ${{ secrets.CODECOV_TOKEN }}
```

Let's add the token explicitly since it seems to be flaky in the other repos without this and there are a few issues when it is not explicitly used 🤔  - https://community.codecov.com/t/upload-issues-unable-to-locate-build-via-github-actions-api/3954

---

I've also once encountered that! That's great we have sucha a simple solution! Thanks @KevFan